pkg/sql/opt/xform/testdata/rules/groupby

```diff
@@ -3662,3 +3662,461 @@ memo (optimized, ~6KB, required=[presentation: d:1,e:2,count:8])
+ │    │    │         ├── columns: c:18!null
+ │    │    │         ├── constraint: /16/17/18/19: [/1/0 - /1/0]
+ │    │    │         ├── ordering: +18
+ │    │    │         └── limit hint: 10.00
+ │    │    └── scan t122638
+ │    │         ├── columns: c:25!null
+ │    │         ├── constraint: /23/24/25/26: [/2/0 - /2/0]
+ │    │         ├── ordering: +25
+ │    │         └── limit hint: 10.00
+ │    └── aggregations
+ │         └── count-rows [as=count_rows:8]
+ └── 10
+
+# Group by next column in index and another column: rule should apply.
+opt expect=SplitGroupByScanIntoUnionScans
+SELECT c, e, count(*) FROM t122638 WHERE b = 0 GROUP BY c, e ORDER BY c, e LIMIT 10
+----
+limit
+ ├── columns: c:3!null e:5!null count:8!null
+ ├── internal-ordering: +3,+5
+ ├── cardinality: [0 - 10]
+ ├── key: (3,5)
+ ├── fd: (3,5)-->(8)
+ ├── ordering: +3,+5
+ ├── group-by (streaming)
+ │    ├── columns: c:3!null e:5!null count_rows:8!null
+ │    ├── grouping columns: c:3!null e:5!null
+ │    ├── internal-ordering: +3,+5
+ │    ├── key: (3,5)
+ │    ├── fd: (3,5)-->(8)
+ │    ├── ordering: +3,+5
+ │    ├── limit hint: 10.00
+ │    ├── union-all
+ │    │    ├── columns: c:3!null e:5!null
+ │    │    ├── left columns: c:32 e:34
+ │    │    ├── right columns: c:25 e:27
+ │    │    ├── ordering: +3,+5
+ │    │    ├── limit hint: 10.00
+ │    │    ├── union-all
+ │    │    │    ├── columns: c:32!null e:34!null
+ │    │    │    ├── left columns: c:11 e:13
+ │    │    │    ├── right columns: c:18 e:20
+ │    │    │    ├── ordering: +32,+34
+ │    │    │    ├── limit hint: 10.00
+ │    │    │    ├── sort (segmented)
+ │    │    │    │    ├── columns: c:11!null e:13!null
+ │    │    │    │    ├── ordering: +11,+13
+ │    │    │    │    ├── limit hint: 10.00
+ │    │    │    │    └── scan t122638
+ │    │    │    │         ├── columns: c:11!null e:13!null
+ │    │    │    │         ├── constraint: /9/10/11/12: [/0/0 - /0/0]
+ │    │    │    │         └── ordering: +11
+ │    │    │    └── sort (segmented)
+ │    │    │         ├── columns: c:18!null e:20!null
+ │    │    │         ├── ordering: +18,+20
+ │    │    │         ├── limit hint: 10.00
+ │    │    │         └── scan t122638
+ │    │    │              ├── columns: c:18!null e:20!null
+ │    │    │              ├── constraint: /16/17/18/19: [/1/0 - /1/0]
+ │    │    │              └── ordering: +18
+ │    │    └── sort (segmented)
+ │    │         ├── columns: c:25!null e:27!null
+ │    │         ├── ordering: +25,+27
+ │    │         ├── limit hint: 10.00
+ │    │         └── scan t122638
+ │    │              ├── columns: c:25!null e:27!null
+ │    │              ├── constraint: /23/24/25/26: [/2/0 - /2/0]
+ │    │              └── ordering: +25
+ │    └── aggregations
+ │         └── count-rows [as=count_rows:8]
+ └── 10
+
+# Group by next column in index, and provides intraOrd: rule should apply.
+opt expect=SplitGroupByScanIntoUnionScans
+SELECT c, array_agg(e)
+FROM (SELECT * FROM t122638 ORDER BY d)
```

Comment:
Interestingly, this seems to be the only way to affect `intraOrd` (not `array_agg(e ORDER BY d)` which would make much more sense).